gateway/util.go

```diff
@@ -157,3 +158,11 @@ func getAPIURL(apiDef apidef.APIDefinition, gwConfig config.Config) string {
 
 	return result.String()
 }
+
+func getAPIDomain(apiDef apidef.APIDefinition) string {
+	domain := apiDef.Domain
+	if apiDef.DomainDisabled {
+		domain = ""
+	}
+	return domain
```

Comment:
can we return empty string to avoid an allocation here 
```suggestion
	if apiDef.DomainDisabled {
		return ""
	}
	return apiDef.Domain
```